atc/db/lock/lock.go

```diff
@@ -24,8 +24,39 @@ const (
 	LockTypeResourceScanning
 	LockTypeJobScheduling
 	LockTypeInMemoryCheckBuildTracking
+	LockTypeResourceGet
+	LockTypeVolumeStreaming
 )
 
+const (
+	factoryDefault = iota
+	factoryVolumeCreating
+	factoryResourceConfigChecking
+	factoryInMemoryCheckBuildTracking
+	factoryBuildTracking
+	factoryJobScheduling
+
+	// FactoryCount must be the last item
+	FactoryCount
+)
+
+func mapLockTypeToFactory(lockType int) int {
+	switch lockType {
```

Comment:
Based on what it decides one lock type to have its own factory e.g. locks count, lock held duration ...?

Reply:
Based on metrics. I just give those locks that are acquired big amount of times dedicate mutexes.